api/agent/agent.go

```diff
@@ -595,10 +628,10 @@ func (a *agent) runHot(ctxArg context.Context, slots chan<- slot, call *call, to
 			}
 
 			done := make(chan struct{})
-			slot := &hotSlot{done, proto, errC, container}
+			qdone := call.slots.queueSlot(&hotSlot{done, proto, errC, container})
 
 			select {
-			case slots <- slot:
+			case <-qdone:
```

Comment:
does this leave us in a position where we have queued the slot, the slot remains queued, yet here we hit the idle timeout and bail but a future call could acquire this slot even though the container is dead?